### PR TITLE
ARCMWDT: Add compiler support for nSIM RMX platform

### PR DESCRIPTION
**Pull request by kokas-a**
Add MetaWare compiler support for nSIM RMX platform

### CODE REVIEW

File: soc/snps/nsim/arc_v/rmx/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,17 @@
 # SPDX-License-Identifier: Apache-2.0
 
+if(COMPILER STREQUAL gcc)
+# GCC:
+
+else ()
+# CCAC:
```
Comment (evgeniy-paltsev) [inline soc/snps/nsim/arc_v/rmx/CMakeLists.txt:7]: Let's drop empty entry for now.

File: soc/snps/nsim/arc_v/rmx/CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+  zephyr_compile_options_ifdef(CONFIG_SOC_SERIES_RMX
+					-av5rmx -Zicsr -Zifencei -Zihintpause -Zba -Zbb
+					-Zbs -Zca -Zcb -Zcmp -Zcmt -Za -Zm -Zicbom)
+
+  zephyr_compile_options(-D__MW_ASM_RV_MACRO__)
```
Comment (evgeniy-paltsev) [inline soc/snps/nsim/arc_v/rmx/CMakeLists.txt:12]: We should move it to toolchain cmake scripts.

File: soc/snps/nsim/arc_v/rmx/CMakeLists.txt (hunk 1 above)
Comment (kokas-a) [reply soc/snps/nsim/arc_v/rmx/CMakeLists.txt]: Fixed

**Review by evgeniy-paltsev — APPROVED** [review]

**Review by abrodkin — APPROVED** [review]

File: soc/snps/nsim/arc_v/rmx/CMakeLists.txt [hunk 3]
```diff
@@ -1,3 +1,12 @@
 # SPDX-License-Identifier: Apache-2.0
 
+if(COMPILER STREQUAL arcmwdt)
+# CCAC:
+  zephyr_compile_options_ifdef(CONFIG_SOC_SERIES_RMX
+					-av5rmx -Zicsr -Zifencei -Zihintpause -Zba -Zbb
+					-Zbs -Zca -Zcb -Zcmp -Zcmt -Za -Zm -Zicbom)
+
+  zephyr_ld_options(-Hlib=rmx100)
```
[the remark this answers is not in the document]
Comment (evgeniy-paltsev) [reply soc/snps/nsim/arc_v/rmx/CMakeLists.txt:9]: For now we distinguish only RMX from RHX series (`-av5rmx` vs `-av5rhx` option)

**Review by ruuddw — APPROVED** [review]

**Pull request merged**